### DISCUSSION TITLE
Detect if a named router view is being used

### DISCUSSION THREAD

**Discussion by eneeio-nm** [vuejs/router#1753] [Help and Questions] [answered]
Hi all,

I would like to know whether there's a way to find out if a named route view is being used.

E.g.

```html
<main>
  <RouterView name="Header" />
  <RouterView name="Sidebar" />
  <RouterView />
</main>
```

In the above example, only some of the routes will include a Header and/or a Sidebar. What I'd like to be able to achieve is to apply a class to the `main` if the named route views are being used.

I've had a look at the `route` object for a route that is using all three router views and that doesn't seem to contain this information.

Any insight into this would be greatly appreciated. Thanks.

**Comment by posva** [comment, marked as answer]
You should be able to find the name in the `route.matched[i].components` object

**Reply by eneeio-nm** [reply]
Yup, that does the trick, thanks @posva
